packages/react-hooks/README.md

```diff
@@ -1,11 +1,301 @@
+## `useFormFields(initialValues, options)`
+`useFormFields` builds binding objects for input fields initialized with `initialValues`. Additionally the hook returns the current values of input fields, the boolean, which tells you if the form has been changed compared to the initial state, a reset function and a `setValue` function. `ResetOnInitialValueChange` boolean decides if the form should be reseted to new inital values (if they have been changed) and hence set the `isDirty` flag to false.
+#### Input
+```javascript
+initialValues = {
+	[key: string]: any
+}
+options = {
+	resetOnInitialValueChange?: boolean
+}
+```
+#### Output
+```javascript
+{
+	bindings: {
+		[key in keyof initialValues]: {
+			{
+				name: string,
+				value: initialValues[key],
+				onChange?: ChangeEventHandler<HTMLInputElement | HTMLTextAreaElement>
+			}
+		}
+	},
+	values: {
+		[key in keyof initialValues]: initialValue[key]
+	},
+	isDirty: boolean,
+	reset: () => void,
+	setValue: (key, value) => void
+}
+```
+#### Example
+```javascript
+import { useFormFields } from "@w2solutions/react-hooks"
+
+const Form = () => {
+	const [color, setColor] = useState("red")
+	const { bindings, values, isDirty, reset, setValue } = useFormFields(
+		{color: color, shape: ''},
+		{resetOnInitialValueChange: true}
+	)
+
+	const submit = useCallback(
+		(event: React.FormEvent<HTMLFormElement>) => {
+			event.preventDefault()
+			setValue("shape",values.shape.toUpperCase())
+			setColor("very " + values.color.toUpperCase())
+			setTimeout(()=>{
+				if (isDirty) {
+					reset()
+				}
+			},2000)
+		}, [values, setValue, reset, isDirty]
+	)
+
+	return (
+		<form onSubmit={(e) => submit(e)}>
+			<input {...bindings.color} type="text"  />
+			<input {...bindings.shape} type="text"/>
+			<button type="submit">Submit</button>
+		</form>
+	)
+}
+```
+>We use `useFormFields` hook to build binding objects for two text input fields: `color` and `shape`. These binding objects contain following properties: `value`, `name` and `onChange` callback if the field is a string input field. The binding objects are placed to the input fields. On the submit event we make the use of the rest of return parameters. We set the color state to upper case and add a string to it. At the same time the initial value of the form field `color` changes and depending on `resetOnInitialValueChange` the values will be reseted or not. `Shape` value will be changed to upper case, but note that its initial value stays still empty and finally we set both fields to initial values after the timeout.
+>
+>Take your time to play with the parameters so that they work the way you want!
+## `usePrevious<SomeType>(value)`
+`usePrevious` returns a previous value of `.current` property of a ref object.
+#### Input
+```javascript
+value: SomeType
```

Comment:
use `T` instead of `SomeType` 

`T` is common for a generic value. `SomeType` could be confusing.